Exercise_5.java

```diff
@@ -1,20 +1,56 @@
-class IterativeQuickSort { 
+// Time Complexity : O(n^2)
+// Space Complexity : O(n) extra array is use dto represent stack
+// Did this code successfully run on Leetcode :  No couldn't find in LeetCode, but executed successfully in IDE
+// Any problem you faced while coding this : I could not fiqure out how to swap without using extra variable
+// Your code here along with comments explaining your approach
+class IterativeQuickSort {
     void swap(int arr[], int i, int j) 
     { 
-	//Try swapping without extra variable 
+	//Try swapping without extra variable
+        int temp;
+        temp = arr[i];
+        arr[i] = arr[j];
+        arr[j] = temp;
     } 
   
     /* This function is same in both iterative and 
        recursive*/
     int partition(int arr[], int l, int h) 
     { 
         //Compare elements and swap.
+        int pivot = arr[high];
+        int ptr1 = low-1;
+        for(int ptr2=low; ptr2<high; ptr2++){
+            if(arr[ptr2] <= pivot){
+                ptr1++;
+                swap(arr,ptr1,ptr2);
+            }
+        }
+        swap(arr,ptr1+1,high);
+        return ptr1+1;
     } 
   
     // Sorts arr[l..h] using iterative QuickSort 
     void QuickSort(int arr[], int l, int h) 

```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.